app/controllers/friend_requests_controller.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class FriendRequestsController < ApplicationController
+  before_action :set_friend_request, only: %i[update destroy]
+
+  def index
+    @sent_requests = current_user.friend_requests_sent.pending
+    @received_requests = current_user.friend_requests_received.pending
+  end
+
+  def create
+    @friend_request = FriendRequest.find_or_initialize_by(sender_id: current_user.id,
```

Comment:
Can we simplify this to just:

```ruby
@friend_request = FriendRequest.new(friend_request_params)
```

or maybe even:

```ruby
@friend_request = current_user.sent_friend_requests.build(receiver_id: params[:friend_request][:receiver_id])
```

Reply:
The mechanism I handled the friend_request, needs to use find_or_initiliaze because after updating request to another state, I don't remove it from the database. So, if there's another attempt from declined/canceled to pending status, this creation logic makes it like the way in an update.